trillian/ctfe/instance.go

```diff
@@ -221,15 +218,25 @@ func SetUpInstance(ctx context.Context, client trillian.TrillianLogClient, cfg *
 		}
 	}
 
-	// Load the private key for this log.
-	var keyProto ptypes.DynamicAny
-	if err := ptypes.UnmarshalAny(cfg.PrivateKey, &keyProto); err != nil {
-		return nil, fmt.Errorf("failed to unmarshal cfg.PrivateKey: %v", err)
-	}
-
-	key, err := keys.NewSigner(ctx, keyProto.Message)
-	if err != nil {
-		return nil, fmt.Errorf("failed to load private key: %v", err)
+	var signer crypto.Signer
+	if !cfg.IsMirror {
+		if cfg.PrivateKey == nil {
+			return nil, errors.New("need to specify PrivateKey")
+		}
+		// Load the private key for this log.
+		var keyProto ptypes.DynamicAny
+		if err := ptypes.UnmarshalAny(cfg.PrivateKey, &keyProto); err != nil {
+			return nil, fmt.Errorf("failed to unmarshal cfg.PrivateKey: %v", err)
+		}
+		var err error
+		if signer, err = keys.NewSigner(ctx, keyProto.Message); err != nil {
+			return nil, fmt.Errorf("failed to load private key: %v", err)
+		}
+	} else {
+		// TODO(pavelkalinnikov): Parse cfg.PublicKey properly.
+		if cfg.PublicKey == nil {
+			return nil, errors.New("need to specify PublicKey")
+		}
```

Comment:
If private key is not required for mirrors is it worth blowing up CTFE here when a config file specifies one for a mirror log - just in the interests of highlighting confusing/wrong/mistaken config files?

Reply:
I agree, will do.